pkg/reconciler/ingress/ingress.go

```diff
@@ -467,6 +467,10 @@ func getLBStatus(gatewayServiceURL string) []v1alpha1.LoadBalancerIngressStatus
 	}
 }
 
-func enableReconcileGateway(ctx context.Context) bool {
-	return config.FromContext(ctx).Network.AutoTLS || config.FromContext(ctx).Istio.ReconcileExternalGateway
+func (r *Reconciler) shouldReconcileTLS(ia *v1alpha1.Ingress) bool {
+	// For the case which the finalizer was added to an Ingress, we should recocnile TLS.
+	// This is because IngressTLS could be added before and then deleted later. We
+	// need to make sure deleting IngressTLS also cleans up the TLS of the Gateway.
+	return (len(ia.GetFinalizers()) != 0 && ia.GetFinalizers()[0] != r.finalizer) ||
```

Comment:
I'd hoist the relevant parts of this into a suitable named variable to reflect the nice comment above.

Reply:
I tried to give a "suitable" name. Hopefully it is actually suitable. :)